src/sagemaker/huggingface/training_compiler/config.py

```diff
@@ -105,3 +107,40 @@ def validate(
                 "transformer_version, tensorflow_version or pytorch_version, and compiler_config."
             )
             raise ValueError(error_helper_string)
+
+        if estimator.distribution:
+            pt_xla_present = "pytorch_xla" in estimator.distribution
+            pt_xla_enabled = estimator.distribution.get("pytorch_xla", {}).get("enabled", False)
+            if pt_xla_enabled:
+                if estimator.tensorflow_version:
+                    error_helper_string = (
+                        "Distribution mechanism 'pytorch_xla' is currently only supported for "
+                        "PyTorch >= 1.11 when Training Compiler is enabled. Received "
+                        "tensorflow_version={} which is unsupported."
+                    )
+                    raise ValueError(error_helper_string.format(estimator.tensorflow_version))
+                elif estimator.pytorch_version:
+                    if Version(estimator.pytorch_version) in SpecifierSet("< 1.11"):
+                        error_helper_string = (
+                            "Distribution mechanism 'pytorch_xla' is currently only supported for "
+                            "PyTorch >= 1.11 when Training Compiler is enabled. Received "
+                            "pytorch_version={} which is unsupported."
+                        )
+                        raise ValueError(error_helper_string.format(estimator.pytorch_version))
+            if not pt_xla_present:
+                if estimator.pytorch_version:
+                    if Version(estimator.pytorch_version) in SpecifierSet(">= 1.11"):
+                        error_helper_string = (
+                            "'pytorch_xla' is the only distribution mechanism currently supported "
```

Comment:
you can still run without specifying distribution, correct?

Reply:
Yes, this is just blocking customers from using other distribution mechanisms like mpi or pytorch ddp.